src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -29,6 +34,99 @@ public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
+			if (OperatingSystem.IsAndroid() && System.IO.File.Exists("/data/local/tmp/ide-launchenv.txt"))
+			{
+				var envVarLines = System.IO.File.ReadAllLines("/data/local/tmp/ide-launchenv.txt");
+
+				environmentVariables = envVarLines
+					.Select(line => line.Split('=', 2))
+					.ToDictionary(parts => parts[0], parts => parts[1]);
+			}
+#endif
+
+			string devTunnelId = environmentVariables["DEVTUNNEL_ID"]?.ToString() ?? string.Empty;
+
+			var variablesToInclude = new HashSet<string>
+			{
+				"ASPNETCORE_ENVIRONMENT",
+				"ASPNETCORE_URLS",
+				"DOTNET_ENVIRONMENT",
+				"DOTNET_LAUNCH_PROFILE",
+				"DOTNET_SYSTEM_CONSOLE_ALLOW_ANSI_COLOR_REDIRECTION"
+			};
+
+			var prefixesToRemove = new List<string>
+			{
+				"ASPNETCORE_",
+				"DOTNET_",
+			};
+
+			List<KeyValuePair<string, string?>> settings = [];
+			foreach (object variableNameObject in environmentVariables.Keys)
+			{
+				string variableName = (string)variableNameObject;
+				if (variablesToInclude.Contains(variableName) || variableName.StartsWith("OTEL_")
+					|| variableName.StartsWith("LOGGING__CONSOLE") || variableName.StartsWith("services__"))
+				{
+					string value = (string)environmentVariables[variableName]!;
+
+					// Normalize the key, matching the logic here:
+					// https://github.dev/dotnet/runtime/blob/main/src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs
+#if NETSTANDARD2_0
+					variableName = variableName.Replace("__", ":");
+#else
+					variableName = variableName.Replace("__", ":", StringComparison.OrdinalIgnoreCase);
+#endif
+
+					// For defined prefixes, add the variable with the prefix removed, matching the logic
+					// in EnvironmentVariablesConfigurationProvider.cs. Also add the variable with the
+					// prefix intact, which matches the normal HostApplicationBuilder behavior, where
+					// there's an EnvironmentVariablesConfigurationProvider added with and another one
+					// without the prefix set.
+					foreach (var prefix in prefixesToRemove)
+					{
+						if (variableName.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+						{
+							settings.Add(new KeyValuePair<string, string?>(variableName, value));
+							variableName = variableName.Substring(prefix.Length);
+							break;
+						}
+					}
+
+					if (!string.IsNullOrEmpty(devTunnelId))
+					{
+						value = ReplaceLocalhost(value, devTunnelId);
+					}
+					settings.Add(new KeyValuePair<string, string?>(variableName, value));
+				}
+			}
+
+			builder.Configuration.AddInMemoryCollection(settings);
+
+			return builder;
+		}
+
+		static string ReplaceLocalhost(string uri, string devTunnelId)
+		{
+			// source format is `http[s]://localhost:[port]`
+			// tunnel format is `http[s]://exciting-tunnel-[port].devtunnels.ms`
+
+			var replacement = Regex.Replace(
+				uri,
+				@"://localhost\:(\d+)(.*)",
```

Comment:
I guess we can't really know if another url was specified here, can we?  How confident are we that localhost is always the thing used.  Should we try to detect anything else? 127.0.0.1?